src/scripts/main.js

```diff
@@ -1,3 +1,399 @@
+  const salary = `$${parseInt(data.salary).toLocaleString('en-US')}`;
+
+  if (cellName.length < 4) {
+    pushNotification(...errorName);
+
+    return;
+  };
+
+  if (age < 18 || age > 90) {
+    pushNotification(...errorAge);
+
+    return;
+  }
+
+  tbody.insertAdjacentHTML('beforeend', `
+    <tr>
+      <td>${cellName}</td>
+      <td>${position}</td>
+      <td>${office}</td>
+      <td>${age}</td>
+      <td>${salary}</td>
+     </tr>
+  `);
+
+  pushNotification(...success);
+
+  form.reset();
+});
+
+tbody.addEventListener('dblclick', e => {
+  const item = e.target.closest('td');
+  const targetCell = item.cellIndex;
+  const currentItem = item.textContent;
+
+  if (!item.querySelector('input') && targetCell < 2) {
+    const input = document.createElement('input');
+
+    item.textContent = '';
+    input.className = ('cell-input');
+    input.value = item.textContent;
+    input.type = 'text';
+
+    item.appendChild(input);
+    input.focus();
+
+    input.addEventListener('keydown', ev => {
+      if (ev.key === 'Enter') {
+        if (input.value.length < 4) {
+          pushNotification(...errorName);
+
+          updateCellInitial(item, currentItem, input);
+
+          return;
+        };
+
+        updateCellCurrent(item, input);
+      }
+    });
+
+    input.addEventListener('blur', ev => {
+      if (!ev.target.value) {
+        ev.target.value = currentItem;
+      }
+
+      if (input.value.length < 4) {
+        pushNotification(...errorName);
+
+        updateCellInitial(item, currentItem, input);
+
+        return;
+      };
+
+      updateCellInitial(item, currentItem, input);
+    });
+  }
+
+  if (!item.querySelector('input') && targetCell === 2) {
+    const select = document.createElement('select');
+
+    select.className = 'cell-input';
+
+    select.insertAdjacentHTML('beforeend', `
+      <option value="Tokyo">Tokyo</option>
+      <option value="Singapore">Singapore</option>
+      <option value="London">London</option>
+      <option value="New York">New York</option>
+      <option value="Edinburgh">Edinburgh</option>
+      <option value="San Francisco">San Francisco</option>
+    `);
+
+    item.textContent = '';
+    item.appendChild(select);
+    select.focus();
+
+    select.addEventListener('change', ev => {
+      updateCellCurrent(item, select);
+    });
+
+    select.addEventListener('blur', ev => {
+      updateCellInitial(item, currentItem, select);
+    });
+
+    select.addEventListener('keydown', ev => {
+      if (ev.key === 'Enter') {
+        updateCellCurrent(item, select);
+      }
+    });
+  }
+
+  if (!item.querySelector('input') && targetCell > 2) {
+    const input = document.createElement('input');
+
+    item.className = 'cell-input';
+    item.textContent = '';
+    input.className = ('cell-input');
+    input.value = item.textContent;
+    input.type = 'number';
+
+    item.appendChild(input);
+    input.focus();
+
+    switch (targetCell) {
+      case 3: {
```

Comment:
please don't use the magic number in a code
you can put this number to the const, please check other places